lib/Rex/Fork/Task.pm

```diff
@@ -56,7 +56,18 @@ sub start {
       exit();
     }
     else {
-      &$func($self);
+      eval { &$func($self) };
+      my $exit_code = $@ ? ($? || 1) : 0;
+      push @PROCESS_LIST, $exit_code;
```

Comment:
Like creating two tasks in the tests, one of them always succeeeding, and the other always failing. Then running each of them and checking if the exit code would be what we expect.

I haven't looked into this deeply yet, it's just an idea. But I guess if we're going to guarantee (and document) this behaviour, it would be very nice to also detect it somehow if we accidentally break this promise.